queue/queue.go

```diff
@@ -56,10 +56,12 @@ func (q *Queue) Close() error {
 }
 
 // Enqueue puts a cid in the queue
-func (q *Queue) Enqueue(cid cid.Cid) {
+func (q *Queue) Enqueue(cid cid.Cid) error {
 	select {
 	case q.enqueue <- cid:
```

Comment:
I'm going to write a followup PR to make this _wait_ for the enqueue to succeed.